src/lib/Form/Type/ChoiceList/Loader/ContentEditTranslationChoiceLoader.php

```diff
@@ -18,54 +18,31 @@
 use Ibexa\Contracts\Core\Repository\Values\Content\Location;
 use Ibexa\Contracts\Core\Repository\Values\User\Limitation;
 
-class ContentEditTranslationChoiceLoader extends BaseChoiceLoader
+final class ContentEditTranslationChoiceLoader extends BaseChoiceLoader
 {
-    private LanguageService $languageService;
-
-    private PermissionResolver $permissionResolver;
-
-    /** @var string[] */
-    private array $languageCodes;
-
-    private ?ContentInfo $contentInfo;
-
-    private LookupLimitationsTransformer $lookupLimitationsTransformer;
-
-    private LocationService $locationService;
-
-    private ?Location $location;
-
     /**
-     * @param \Ibexa\Contracts\Core\Repository\LanguageService $languageService
-     * @param \Ibexa\Contracts\Core\Repository\PermissionResolver $permissionResolver
-     * @param \Ibexa\Contracts\Core\Repository\Values\Content\ContentInfo $contentInfo
-     * @param \Ibexa\AdminUi\Permission\LookupLimitationsTransformer $lookupLimitationsTransformer
      * @param string[] $languageCodes
      */
     public function __construct(
-        LanguageService $languageService,
-        PermissionResolver $permissionResolver,
-        ?ContentInfo $contentInfo,
-        LookupLimitationsTransformer $lookupLimitationsTransformer,
-        array $languageCodes,
-        LocationService $locationService,
-        ?Location $location
+        private readonly LanguageService $languageService,
+        private readonly PermissionResolver $permissionResolver,
+        private readonly ?ContentInfo $contentInfo,
```

Comment:
Those will throw deprecation in php 8.4, 
dont know if this is a time to fix those tho as this involves also order of arguments change (but it shouldnt be big as this is service, right?) so it is up to you.